lib/teleterm/clusters/cluster_gateways.go

```diff
@@ -139,18 +145,36 @@ func (c *Cluster) createKubeGateway(ctx context.Context, params CreateGatewayPar
 	return gw, trace.Wrap(err)
 }
 
-// ReissueGatewayCerts reissues certificate for provided gateway.
-func (c *Cluster) ReissueGatewayCerts(ctx context.Context, g gateway.Gateway) error {
+// ReissueGatewayCerts reissues certificate for the provided gateway.
+//
+// At the moment, kube gateways reload their certs in memory while db gateways use the old approach
+// of saving a cert to disk and only then loading it to memory.
+// TODO(ravicious): Refactor db gateways to reload cert in memory and support MFA.
+func (c *Cluster) ReissueGatewayCerts(ctx context.Context, g gateway.Gateway, mfaPrompt mfa.Prompt) (tls.Certificate, error) {
 	switch {
 	case g.TargetURI().IsDB():
 		db, err := gateway.AsDatabase(g)
 		if err != nil {
-			return trace.Wrap(err)
+			return tls.Certificate{}, trace.Wrap(err)
+		}
+		err = c.reissueDBCerts(ctx, db.RouteToDatabase())
+		if err != nil {
+			return tls.Certificate{}, trace.Wrap(err)
 		}
-		return trace.Wrap(c.reissueDBCerts(ctx, db.RouteToDatabase()))
+
+		// db gateways still store certs on disk, so they need to load it after reissue.
+		err = g.ReloadCert()
```

Comment:
`daemon.Service.reissueGatewayCerts` used to call `Cluster.ReissueGatewayCerts` followed by `gateway.base.ReloadCert`. Calling it for kube gateways no longer makes sense, so the call to `ReloadCert` has been moved here.

Perhaps I should just rename `ReloadCert` to `ReloadCertFromDisk`.